src/operations/indexes.ts

```diff
@@ -51,14 +51,17 @@ const VALID_INDEX_OPTIONS = new Set([
 
 /** @public */
 export type IndexDirection = -1 | 1 | '2d' | '2dsphere' | 'text' | 'geoHaystack' | number;
-
+function isIndexDirection(x: any): x is IndexDirection {
+  return (
+    typeof x === 'number' || x === '2d' || x === '2dsphere' || x === 'text' || x === 'geoHaystack'
+  );
+}
 /** @public */
 export type IndexSpecification = OneOrMore<
   | string
   | [string, IndexDirection]
   | { [key: string]: IndexDirection }
```

Comment:
Of the three we're addressing here should we pull them each out and try and make this PR only about the refactor (i.e. why this bug but not the others)? (I think the tuple one might be difficult but we can see).

Reply:
what "three" are you referencing here? three bugs?

Reply:
- Map in TS
- Key order FLE
- Tuple parsing issue
- Type strictness on this nesting of array (one or more)
- Type strictness for createIndexes aligned with createIndex

Reply:
I think whether we pull these out into separate PRs/fixes or not, we *should* make sure that the PR description accurately reflects whatever fixes are included and that we are *super* clear in our release notes about the changes in behavior. It's also worth noting that the types in the documentation won't get updated on a patch, unless we manually regenerate the 4.8, which could be confusing. I think because of the scope of the changes here, this set of improvements that's coming in as a bug fix is better marked as a `feat:`

Reply:
Oh and let's triple check to make sure that every single one of those things has full test coverage

Reply:
Collected a summary in the description and between the type/fle/indexes.test.ts additions I see coverage for each point.